SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -1308,4 +1316,63 @@ extension PositioningViewController {
     }
 }
 
-
+extension PositioningViewController {
+    func initializeFloorChangerMarker() {
+        let icon = self.scaledImage(
+            image: UIImage(named: "change_floor")!,
+            scaledToSize: CGSize(width: 40.0, height: 40.0)
+        )
+        let markerView = UIImageView(image: icon)
+        self.changeOfFloorMarker.iconView = markerView
+        self.changeOfFloorMarker.position = CLLocationCoordinate2D(latitude: 0, longitude: 0)
+        resetFloorMarker()
+    }
+    
+    func showFloorChangeMarker(position: CLLocationCoordinate2D) {
+        self.changeOfFloorMarker.position = position
+        self.changeOfFloorMarker.map = self.mapView
+    }
+    
+    func scaledImage(image:UIImage, scaledToSize newSize:CGSize) -> UIImage{
+        UIGraphicsBeginImageContextWithOptions(newSize, false, 0.0)
+        image.draw(in: CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height))
+        let newImage:UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+        UIGraphicsEndImageContext()
+        return newImage
+    }
+    
+    func changePositionFloorMarkerOrResetFloorMarker(segments: Array<SITRouteSegment>, selectedFloor: SITFloor) {
```

Comment:
Esto o que fai realmente se non me trabuco é pintar os cambios de planta que hai para un piso, non?
drawChangeOfFloorMarker(floor: segments)?

Reply:
De todas formas se nunha mesma planta houbese máis dun cambio de planta por que a ruta sube a outro piso e despois baixa quedaste co ultimo non?, se só se pinta ún posiblmente debería ser o que che quede máis cerca na ruta